hermes-frontend/src/test/groovy/pl/allegro/tech/hermes/frontend/producer/kafka/KafkaMessagesAsyncProducerTest.groovy

```diff
@@ -0,0 +1,79 @@
+package pl.allegro.tech.hermes.frontend.producer.kafka
+
+import pl.allegro.tech.hermes.frontend.metric.CachedTopic
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessageProducer
+import pl.allegro.tech.hermes.frontend.producer.BrokerMessagesProducingResults
+import pl.allegro.tech.hermes.frontend.publishing.PublishingCallback
+import pl.allegro.tech.hermes.frontend.publishing.message.JsonMessage
+import pl.allegro.tech.hermes.frontend.publishing.message.Message
+import spock.lang.Shared
+import spock.lang.Specification
+
+import static com.google.common.base.Charsets.UTF_8
+import static pl.allegro.tech.hermes.frontend.server.CachedTopicsTestHelper.cachedTopic
+
+class KafkaMessagesAsyncProducerTest extends Specification {
+
+    @Shared
+    CachedTopic cachedTopic = cachedTopic("group.test")
+
+    def "should successfully publish messages async if publishing single message on broker is available"() {
```

Comment:
I think you don't need to use `async` word here ;) If you would like to change the implementation you don't have to change the test name

Reply:
The name of the test is quite puzzling :) Mayme just `should successfully publish messages if broker is available`